CGAT/scripts/runGSEA.py

```diff
@@ -0,0 +1,1619 @@
+                str(C) +
+                " gene sets." +
+                "\n")
+        f.write("The remaining " + str(C - len(EX)) +
+                " gene sets were used in the analysis." + "\n")
+        f.write(
+            "List of gene sets used and their sizes (restricted to features in the specified dataset) are mentioned here:" +
+            "\n")
+        f.write(
+            "\n" +
+            "NAME" +
+            "\t" +
+            "ORIGINAL SIZE" +
+            "\t" +
+            "AFTER RESTRICTING TO EXPRESSION DATASET" +
+            "\t" +
+            "STATUS" +
+            "\n")
+        for i in range(0, len(geneset_indicator)):
+            if(geneset_indicator[i] < 0):
+                f.write(EX[abs(geneset_indicator[i]) -
+                           1][0] +
+                        "\t" +
+                        str(EX[abs(geneset_indicator[i]) -
+                               1][2]) +
+                        "\t" +
+                        "\t" +
+                        " Rejected!" +
+                        "\n")
+            else:
+                f.write(IN[geneset_indicator[i] -
+                           1][0] +
+                        "\t" +
+                        str(IN[geneset_indicator[i] -
+                               1][2]) +
+                        "\t" +
+                        str(IN[geneset_indicator[i] -
+                               1][3]) +
+                        '  ' +
+                        "\n")
+    f.close()
+    return
+
+
+def generate_leading_edge_m(
+        I_U,
+        I_D,
+        f_U,
+        f_D,
+        IN_PRO,
+        hm,
+        dict_new,
+        dfp,
+        ufp,
+        matr,
+        OE,
+        OEI,
+        id_for_leading,
+        store_gene_leading_matrix,
+        gene_set_a,
+        store_gene_leading_info):
+    #####################################################################
+    # I am writing this section for leading edge analysis. It is necessary to describe each and every parameter
+    # here (just for myself).
+    # I_U,I_D and f_U,f_D are sorted upregulated and downregulated indexes (on the basis of NES values) and
+    # coreesponding FDR values.
+    # IN_PRO is list of included geneset "in_list"
+    # dict_new is dictionary format of expression dataset "ind_dict"
+    # gene_set_a is stored included gene set "GG"
+    # id_for_leading is id of expression dataset. I am using array format "id_new"
+    # Apart from that, I am using store_gene_leading_info as SGLI, original_es_index as OEI, original_es as OE
+    # I am also using up_for_plot and down_for_plot as ufp and dfp, which stores information for retrieving
+    # original index in original_nes.
+    # ch is choice of user FDR and hm is selected number of genes for leading edge analysis
+    # matr is temp matrix, used by me earlier.
+    # This section is little bit complex.
+    #######################################################################
```

Comment:
Should this be a `python docstring` instead?

For reference:
* http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
* https://www.python.org/dev/peps/pep-0257/